p2p/simulations/network.go

```diff
@@ -718,17 +721,102 @@ func (net *Network) Load(snap *Snapshot) error {
 			return err
 		}
 	}
+
+	// Prepare connection events counter.
+	allConnected := make(chan struct{})     // closed when all connections are established
+	eventLoopStarted := make(chan struct{}) // ensures that event loop is started before it is closed
+	done := make(chan struct{})             // ensures that the event loop goroutine is terminated
+	defer close(done)
+
+	go func() {
+		// Subscribe to event channel.
+		events := make(chan *Event)
+		sub := net.Events().Subscribe(events)
+		defer sub.Unsubscribe()
+
+		// Expected number of connections.
+		total := len(snap.Conns)
+		// Set of all established connections from the snapshot, not other connections.
+		// Key array element 0 is the connection One field value, and element 1 connection Other field.
+		connections := make(map[[2]enode.ID]struct{}, total)
+
+		// once is a closed channel that is read in the event loop below
+		// only once.
+		// It ensures that eventLoopStarted is closed which signals that
+		// it is safe to call connect method on the network without the
+		// possibility to miss a few first connection events.
+		once := make(chan struct{})
+		// Close once channel so that it can be read from in the event loop.
+		close(once)
+
+		for {
+			select {
+			case e := <-events:
+				// Detect only connection events.
+				if e.Type != EventTypeConn {
```

Comment:
Should we not disregard the `Control` events aswell?

Reply:
We can, but could you elaborate what would we protect agains?

Reply:
I've seen that that for connections in p2p layer there _are_ or _can be_ **two** conn events being sent, the first of which will be fired even though the node object at that time will have `Conn.Up = false` for that connection in its state.

Reply:
Yes, I see, I forgot about this `Up = false` events. Thank you, that is a great suggestion. I've updated the code.